src/App.tsx

```diff
@@ -1,18 +1,81 @@
-import styled from '@emotion/styled';
+import React, { createContext, useState, useEffect, useMemo } from 'react';
+import { Global } from '@emotion/react';
+import resetStyles from '@styles/resetStyles';
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  Navigate,
+} from 'react-router-dom';
+import Layout from '@components/Layout/Layout';
+import Main from '@pages/Main/Main';
+import ThemePage from '@pages/ThemePage/ThemePage';
+import LoginPage from '@pages/Login/LoginPage';
+import MyAccountPage from '@pages/MyAccount/MyAccountPage';
+import { ROUTE_PATHS } from './constants';
+
+const AuthContext = createContext({
+  isLoggedIn: false,
+  // eslint-disable-next-line no-unused-vars
+  login: (id: string) => {},
+  logout: () => {},
+});
 
 const App = () => {
-  const name = 'Josh Perez';
+  const [isLoggedIn, setIsLoggedIn] = useState(
+    !!sessionStorage.getItem('authToken'),
+  );
```

Comment:
```suggestion
  const [isLoggedIn, setIsLoggedIn] = useState(() => {
    return sessionStorage.getItem('authToken');
  });
```

리렌더링 될 때 마다 세션 스토리지의 값을 불러올거라 useState 인자로 함수를 전달해서 지연 초기화 하는 방법으로 개선 가능합니다.

- [useState(initialState)](https://ko.react.dev/reference/react/useState#usestate)

> 함수를 initialState로 전달하면 이를 초기화 함수로 취급합니다. 이 함수는 순수해야 하고 인수를 받지 않아야 하며 반드시 어떤 값을 반환해야 합니다. React는 컴포넌트를 초기화할 때 초기화 함수를 호출하고, 그 반환값을 초기 state로 저장합니다. 아래 예시를 참고하세요.